msal/public_client_application.go

```diff
@@ -129,4 +129,5 @@ type AcquireTokenByDeviceCodeOptions struct {
 type AcquireTokenByAuthCodeOptions struct {
 	Code          string
 	CodeChallenge string
+	RedirectURI   string
```

Comment:
@jmprieur can you please comment on this?  I explicitly removed this in https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/68 per your request.

Reply:
I am not the original requester of that change, although our handles look similar. In my case, I was trying to use a token-fetch as an explicit call back to MS instead of implicit-token flow. As per the documentation, implicit token on auth-code flow is less secure and they recommend webapps fetch it back from MS. When we call back, this redirectURI is required, and matching original auth request.

Reply:
The recommendation is for apps to use the `https://login.microsoftonline.com/common/oauth2/nativeclient` reply URL, this is what I did when testing the change and it worked for me.  See the conversation [here](https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/68#discussion_r516579950).  I'm far from an expert here though so somebody from the MSAL team needs to comment further.

MSAL for other languages don't expose a redirect URI parameter.

Reply:
@jmprieur I don't quite understand this statement that you made in #68 
> but MSAL stops the redirect

My use-case is a webapp that needs to redirect a user to MS auth servers and then back to the site. It would be what you consider a confidential-app and I am specifically choosing not to use 'implicit-token' checkbox in the App settings since that is not recommended.

After the user accepts the consent, the browser needs to be redirected back to the site, so I do need to supply a real redirect-url in addition to make it match with one of the configured urls on the app settings.  If you are saying MS will redirect back to my real site even if I specify `login.microsoftonline.com`, then I am mistaken, I have not tried that.

Now the specific fix is about making it configurable, so that I can pass the same value back when I fetch the token.
My app is not opensource, but it is very similar to your sample

First place I need to use the redirect-url
https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/test/devapps/confidential_auth_code_sample.go#L29

Second place I need to use the redirect-url (this fix)
https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/test/devapps/confidential_auth_code_sample.go#L61

Reply:
After discussion we do need to add this back.  However we shouldn't place this in the options struct as it makes it appear optional when it should be user-specified (falling back to a default value is incorrect).  Can you please move this out of the options struct and make it a parameter so it's clear the caller must supply a value?